ssz/sedes/base.py

```diff
@@ -0,0 +1,149 @@
+import abc
+
+from typing import (
+    Generic,
+    Tuple,
+    TypeVar,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+
+
+T = TypeVar("T")
+S = TypeVar("S")
+
+
+class BaseSedes(abc.ABC, Generic[T, S]):
+
+    @abc.abstractmethod
+    def serialize(self, value: T) -> bytes:
+        raise NotImplementedError(
```

Comment:
I remember a wise man told me that we don't need `raise NotImplementedError()` if we use `abstractmethod`. Just `pass`.

Reply:
Correct, `ABC` ensures classes can not be instantiated with abstract methods so the exception will never be hit and a `pass` is just a bit shorter to type.